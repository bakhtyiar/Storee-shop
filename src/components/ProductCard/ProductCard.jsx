import React, {useContext} from 'react';
import imageIcon from '../../assets/img/icons/image/icons8-image-48.png';
import {routes} from '../../utils/constants';
import {Link} from 'react-router-dom';
import {Button} from "react-bootstrap";
import {RootContext} from "../../contexts/root-context/root-context";

const ProductCard = ({item}) => {
    const { cartState: { onAddToCart } } = useContext(RootContext);

    return (
        <Link className='px-2 py-3 border rounded d-flex flex-column align-items-start text-decoration-none text-black' to={`${routes.product.path}/${item.id}`} key={item.id}>
            <div className={'bg-image rounded align-self-center'}
                 style={{
                     height: '128px',
                     width: '100%',
                     backgroundImage: `url(${item.thumbnail || imageIcon})`,
                     backgroundPosition: 'center',
                     backgroundSize: 'contain',
                     backgroundRepeat: 'no-repeat'
                 }}/>
            <h6 className='mt-2'>{item.title}</h6>
            <p>Price: ${item.price}</p>
            <Button
                variant='dark'
                className='w-100'
                onClick={ (e) => {
                    e.preventDefault();
                    onAddToCart(item.id);
                }}
            >
                <i className="bi bi-bag-plus"></i>
                {' '}
                Add to cart
            </Button>
        </Link>
    );
}

export default ProductCard;