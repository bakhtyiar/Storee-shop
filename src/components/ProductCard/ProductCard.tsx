import React, {useContext} from 'react';
import imageIcon from '../../assets/img/icons/image/icons8-image-48.png';
import {routes} from '../../utils/constants';
import {Link} from 'react-router-dom';
import {Badge, Button} from "react-bootstrap";
import {RootContext} from "../../contexts/root-context/root-context";

const ProductCard = ({
    item
}: any) => {
    const {cartState: {onAddToCart}} = useContext(RootContext);

    return (
        <Link className='px-2 py-3 border rounded d-flex flex-column align-items-start text-decoration-none text-black'
              to={`${routes.product.path}/${item.id}`} key={item.id} data-testid={"product-card"}>
            
            <div className={'bg-image rounded align-self-center'}
                 style={{
                     height: '128px',
                     width: '100%',
                     backgroundImage: `url(${item.thumbnail || imageIcon})`,
                     backgroundPosition: 'center',
                     backgroundSize: 'contain',
                     backgroundRepeat: 'no-repeat'
                 }}/>
            
            <p className={'mb-0'}><i className="bi bi-star-fill text-warning"></i> <small>{item.rating}</small></p>
            
            <p className='mb-0'><b>{item.title}</b></p>
            
            <div>
                
                <small><del>${item.price}</del></small>
                
                <br/>
                
                <p className={'d-flex'}><big><b>${Math.round(item.price * (100 - item.discountPercentage) / 100)}</b></big> <Badge className={'align-self-start ms-1'} bg="warning" text="dark">{item.discountPercentage}%</Badge></p>
            </div>
            
            <Button
                variant='dark'
                className='w-100'
                onClick={(e) => {
                    e.preventDefault();
                    onAddToCart(item.id);
                }}
                data-testid="add-to-card-button"
            >
                
                <i className="bi bi-bag-plus"></i>
                {' '}
                Add to cart
            </Button>
        </Link>
    );
}

export default ProductCard;