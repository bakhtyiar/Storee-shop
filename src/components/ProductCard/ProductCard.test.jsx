import {screen} from "@testing-library/react";
import {renderWithRouter} from "../../tests/helpers/renderWithRouter";
import ProductCard from "./ProductCard";
//todo: write interactive tests
describe("Basic tests", () => {
    test("Match snapshot", () => {
        renderWithRouter(<ProductCard item={{id: 123, }}/>);
        expect(screen.getByTestId("product-card")).toMatchSnapshot();
    })
})