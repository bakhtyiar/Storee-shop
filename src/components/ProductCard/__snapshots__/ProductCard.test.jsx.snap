// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic tests Match snapshot 1`] = `
<a
  class="px-2 py-3 border rounded d-flex flex-column align-items-start text-decoration-none text-black"
  data-testid="product-card"
  href="/product/123"
>
  <div
    class="bg-image rounded align-self-center"
    style="height: 128px; width: 100%; background-image: url(icons8-image-48.png); background-position: center; background-size: contain; background-repeat: no-repeat;"
  />
  <p
    class="mb-0"
  >
    <i
      class="bi bi-star-fill text-warning"
    />
     
    <small />
  </p>
  <p
    class="mb-0"
  >
    <b />
  </p>
  <div>
    <small>
      <strike>
        $
      </strike>
    </small>
    <br />
    <p
      class="d-flex"
    >
      <big>
        <b>
          $
          NaN
        </b>
      </big>
       
      <span
        class="align-self-start ms-1 badge text-dark bg-warning"
      >
        %
      </span>
    </p>
  </div>
  <button
    class="w-100 btn btn-dark"
    type="button"
  >
    <i
      class="bi bi-bag-plus"
    />
     
    Add to cart
  </button>
</a>
`;
